Codec/Archive/Tar/Unpack.hs

```diff
@@ -74,7 +86,7 @@ unpack baseDir entries = unpackEntries [] (checkSecurity entries)
     unpackEntries _     (Fail err)      = either throwIO throwIO err
     unpackEntries links Done            = return links
     unpackEntries links (Next entry es) = case entryContent entry of
-      NormalFile file _ -> extractFile path file mtime
+      NormalFile file _ -> extractFile entry path file mtime
```

Comment:
If `extractFile` uses only `entryPermissions entry`, could we please make it explicit by passing this field only instead of whole `entry`?